diff --git a/admin_menu.module b/admin_menu.module
index 8a8dee1..925a9ea 100755
--- a/admin_menu.module
+++ b/admin_menu.module
@@ -160,7 +160,7 @@ function admin_menu_page_build(&$page) {
   if (strpos($_GET['q'], 'js/') === 0) {
     return;
   }
-  global $user, $language;
+  global $user, $language, $_domain;
   $path = drupal_get_path('module', 'admin_menu');
 
   $page['page_bottom']['admin_menu'] = array(
@@ -191,6 +191,12 @@ function admin_menu_page_build(&$page) {
   // If the client supports JavaScript and we have a cached menu for the current
   // user, only output the hash for the client-side HTTP cache callback URL.
   $cid = 'admin_menu:' . $user->uid . ':' . session_id() . ':' . $language->language;
+  // Better integration with DomainAccess.
+  // Add domain_id to $cid.
+  if (isset($_domain['domain_id'])) {
+    $cid .= ':' . $_domain['domain_id'];
+  }
+
   if (!$complete && !empty($_COOKIE['has_js']) && ($hash = admin_menu_cache_get($cid))) {
     $settings['hash'] = $hash;
     // The base path to use for cache requests depends on whether clean URLs
@@ -457,10 +463,15 @@ function admin_menu_session_count($timestamp = 0, $anonymous = TRUE) {
  *   page.
  */
 function admin_menu_output($complete = FALSE) {
-  global $user, $language;
+  global $user, $language, $_domain;
 
   $cache_server_enabled = !$complete && variable_get('admin_menu_cache_server', TRUE);
   $cid = 'admin_menu:' . $user->uid . ':' . session_id() . ':' . $language->language;
+  // Better integration with DomainAccess.
+  // Add domain_id to $cid.
+  if (isset($_domain['domain_id'])) {
+    $cid .= ':' . $_domain['domain_id'];
+  }
 
   // Try to load and output administration menu from server-side cache.
   // @todo Duplicates the page cache? Page cache ID contains the hash that is
