diff --git a/password_policy.module b/password_policy.module
index 6de50f5..3db2a08 100644
--- a/password_policy.module
+++ b/password_policy.module
@@ -294,7 +294,7 @@ function password_policy_user_presave(&$edit, $account, $category) {
     // And if this is not a newly created user...
     if (!$account->is_new) {
       // And if the pass value is not the same as before...
-      if ($edit['pass'] != $account->pass) {
+      if ($edit['pass'] != $account->original->pass) {
         // Then store the password hash to history.
         _password_policy_store_password($account->uid, $edit['pass']);
       }
